Sim/SimpleFtl/SimpleFtl.cpp

```diff
@@ -37,13 +37,28 @@ bool SimpleFtl::SetSectorInfo(const SectorInfo &sectorInfo)
         * geometry.BlocksPerDevice * geometry.PagesPerBlock * _SectorsPerPage;
     SimpleFtlTranslation::SetSectorSize(sectorInfo.SectorSizeInBit);
 
-    _SectorsPerSegment = _SectorsPerPage;
+    // The buffer size must be larger than maxBlocksPerCommand blocks' size to support read-modify-write
+    constexpr U8 maxBlocksPerCommand = 8;
+    U32 blockSizeInBytes = geometry.PagesPerBlock * geometry.BytesPerPage;
+    assert(_BufferHal->GetBufferMaxSizeInBytes() >= blockSizeInBytes * maxBlocksPerCommand);
+    _BlockBuffers = new Buffer[geometry.PagesPerBlock * maxBlocksPerCommand];

```

Comment:
Is dynamic allocation temporary? How do we want to handle it long term?

Reply:
Yes, it is temporary. I haven't found any ways to avoid any dynamic allocation currently.

Reply:
So I added to support a large buffer for a block to remove the dynamic allocation.